app/views/alaveteli_pro/info_requests/_embargo_form.html.erb

```diff
@@ -1,45 +1,32 @@
-<% if info_request.embargo %>
+<% if info_request.embargo || can?(:create_embargo, info_request) %>
 <div class="sidebar__section update-embargo">
   <h2 class="embargo-sidebar-heading">
+    <% if info_request.embargo %>
     <i class="embargo-indicator embargo-indicator--small"
        title="<%= _('Private until {{date}}',
                     :date => simple_date(info_request.embargo.publish_at,
                                          :format => :text)
                    ) %>"></i>
+    <% end %>
     <%= _("Privacy") %>
   </h2>
   <label class="houdini-label" for="input1"><%= _("Change privacy") %></label>
   <input class="houdini-input" type="checkbox" id="input1">
   <div class="houdini-target extend-embargo-sidebar">
-    <%= form_for(
-          [AlaveteliPro::EmbargoExtension.new(embargo: info_request.embargo)],
-          html: {class: 'js-embargo-form'}) do |f| %>
-      <%= render partial: "alaveteli_pro/info_requests/embargo_info",
-                 locals: {embargo: info_request.embargo, tense: :present} %>
-      <%= f.hidden_field :embargo_id %>
-
-      <p>
-        <label class="form_label"
-               for="alaveteli_pro_embargo_extension_extension_duration">
-          <%= _('Keep private for a further:') %>
-        </label>
-        <%= f.select :extension_duration,
-                     options_for_select(
-                       embargo_extension_options(info_request.embargo)),
-                     {},
-                     { class: 'js-embargo-duration' } %>
-        <input type="submit"
-               value="<%= _("Update") %>"
-               class="embargo__submit js-embargo-submit">
-      </p>
+    <% if info_request.embargo %>
+      <%= render partial: "alaveteli_pro/info_requests/embargo_extension_form",
+                 locals: {info_request: info_request} %>
+      <%= button_to _("Publish request"),
+                    alaveteli_pro_embargo_path(info_request.embargo),
+                    method: :delete,
+                    data: {
+                      confirm: _("Are you sure you want to publish " \
+                                 "this request?")
+                    } %>
+    <% else %>
```

Comment:
This `else` conditional will render the create form on batched requests - but when the form is submitted it raises `PermissionDenied` exception [here](https://github.com/mysociety/alaveteli/pull/4287/files#diff-6b4fa2fa47d11a529cdf2a0b0ff63004R15).

While we could prevent render here using an `elsif` and checking to see if `info_request.info_request_batch_id.nil?` I think a nicer solution would be to move this check into the `:create_embargo` ability

Reply:
good catch, cheers!